packages/tools/src/utilities/math/polyline/intersectPolyline.ts

```diff
@@ -0,0 +1,30 @@
+import { Types } from '@cornerstonejs/core';
+import getFirstLineSegmentIntersectionIndexes from './getFirstLineSegmentIntersectionIndexes';
+
+function intersectPolyline(
+  sourcePolyline: Types.Point2[],
+  targetPolyline: Types.Point2[]
+): boolean {
+  // Naive way to detect intersection between polylines in O(n^2).
+  // TODO: Implement Bentley Ottmann sweep line algorithm or maybe some
+  // algorithm that uses r-tree may make it run faster
+  for (let i = 0, sourceLen = sourcePolyline.length; i < sourceLen; i++) {
+    const sourceP1 = sourcePolyline[i];
+    const sourceP2Index = i === sourceLen - 1 ? 0 : i + 1;
+    const sourceP2 = sourcePolyline[sourceP2Index];
+
+    const intersectionPointIndexes = getFirstLineSegmentIntersectionIndexes(
+      targetPolyline,
+      sourceP1,
+      sourceP2
+    );
+
+    if (intersectionPointIndexes?.length === 2) {
+      return true;
+    }
+  }
```

Comment:
handle the first and last one outside the loop so that you remove that check inside each iteration


for (let i = 0, sourceLen = sourcePolyline.length; i < sourceLen; i++) {
    const sourceP1 = sourcePolyline[i];
    const sourceP2 = sourcePolyline[i+1]

    const intersectionPointIndexes = getFirstLineSegmentIntersectionIndexes(
      targetPolyline,
      sourceP1,
      sourceP2
    );

    if (intersectionPointIndexes?.length === 2) {
      return true;
    }
  }

  const sourceP1 = sourcePolyline[sourceLen - 1];
const sourceP2 = sourcePolyline[0];


...

Reply:
Why? Performance? I cloned/refactored the function and I do not see any difference when calculating the intersection between two polylines with 5k points each (25M intersections calculated for the naive implementation). I wouldn't like to duplicate the code for no improvement.

![Screenshot_20240124_151915](https://github.com/cornerstonejs/cornerstone3D/assets/931820/55e6f925-4a2e-47bb-ab02-480c77036caf)

![Screenshot_20240124_151820](https://github.com/cornerstonejs/cornerstone3D/assets/931820/2b8307ad-1200-460b-a9a7-24bea5ddfade)